libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -51,251 +51,72 @@ void ArrayUtils::copyArrayToStorage(ArrayType const& _targetType, ArrayType cons
 
 	Type const* targetBaseType = _targetType.baseType();
 	Type const* sourceBaseType = _sourceType.baseType();
-
-	// TODO unroll loop for small sizes
-
-	bool sourceIsStorage = _sourceType.location() == DataLocation::Storage;
 	bool fromCalldata = _sourceType.location() == DataLocation::CallData;
-	bool directCopy = sourceIsStorage && sourceBaseType->isValueType() && *sourceBaseType == *targetBaseType;
-	bool haveByteOffsetSource = !directCopy && sourceIsStorage && sourceBaseType->storageBytes() <= 16;
-	bool haveByteOffsetTarget = !directCopy && targetBaseType->storageBytes() <= 16;
-	unsigned byteOffsetSize = (haveByteOffsetSource ? 1u : 0u) + (haveByteOffsetTarget ? 1u : 0u);
+	bool hasDynamicCalldataLength = fromCalldata && _sourceType.isDynamicallySized();
 
-	// stack: source_ref [source_length] target_ref
-	// store target_ref
+	// TODO unroll loop for small sizes
 	for (unsigned i = _sourceType.sizeOnStack(); i > 0; --i)
 		m_context << swapInstruction(i);
```

Comment:
Please put this TODO back where it was. The change is misleading, making it look as if it was about this small swap loop, but that would make no sense. I think it was about unrolling the storage copying loops when arrays are small.

TBH we'll never doing this after all, at least not in the old codegen, but it's still useful to see that originally this was the plan.

Reply:
Oh right, I moved things around and that ended up there. I will move it back.